### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by janpio**
- Greatly improved introduction
- Added "What does it do?" that explains all the individual things it does
- Resorted and grouped the CLI parameters
- Added some common use cases
- Added Table of Contents

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,40 +1,183 @@
...
+## Installation
+
+```
+npm install -g cordova-paramedic
+
+# or
+
+git clone https://github.com/apache/cordova-paramedic
+# when cloning, you have to run `npm link` inside the checkout
+# or replace all occurences of `cordova-paramedic` in commands with `cordova-paramedic/main.js`
```
Comment (alsorokin) [inline README.md]: My memory may be deceiving me, but isn't it `node cordova-paramedic/main.js`?

Comment (janpio) [reply README.md]: Good point, it probably is on Windows in any case. 

Linux and Mac might be happy with 

https://github.com/apache/cordova-paramedic/blob/336ccca9f8d6290968f27a8e527084a102256aef/main.js#L1

Correct?

File: README.md [hunk 2]
```diff
@@ -1,40 +1,183 @@
...
+<!--<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>-->
 
 ## Supported Cordova Platforms
 
 - Android
 - iOS
-- Windows (Windows 8.1, Windows 10 UWP phone,tablet,desktop)
+- Windows
 - Browser
-- OSX
```
Comment (erisu) [inline README.md]: Confirm?

Comment (janpio) [reply README.md]: From the code that documents the parameters:  `"--platform PLATFORM : the platform id. Currently supports 'ios', 'browser', 'windows', 'android'.\n" +`. There is no mention of `osx` at all.

I created https://github.com/apache/cordova-paramedic/issues/89 as this should definitely be supported though for a core platform.

File: README.md [hunk 3]
```diff
@@ -1,40 +1,183 @@
...
+## Installation
+
+```
+npm install -g cordova-paramedic
+
+# or
+
+git clone https://github.com/apache/cordova-paramedic
+# when cloning, you have to run `npm link` inside the checkout
+# or replace all occurences of `cordova-paramedic` in commands with `cordova-paramedic/main.js`
```
Comment (janpio) [inline README.md]:
```suggestion
# or replace all occurences of `cordova-paramedic` in commands with `cordova-paramedic/main.js`
# or `node cordova-paramedic/main.js` (on Windows)
```

**Pull request merged**